test/unit/lib/plugins/aws/package/compile/events/mq.test.js

```diff
@@ -0,0 +1,218 @@
+                  },
+                },
+              ],
+            },
+            other: {
+              events: [
+                {
+                  mq: {
+                    queue,
+                    arn: brokerArn,
+                    accessConfigurations: { basicAuth: basicAuthArn },
+                    batchSize,
+                    enabled,
+                  },
+                },
+              ],
+            },
+          },
+        },
+        cliArgs: ['package'],
+      });
+      naming = awsNaming;
+      minimalEventSourceMappingResource =
+        cfTemplate.Resources[naming.getMQEventLogicalId('foo', queue)];
+      allParamsEventSourceMappingResource =
+        cfTemplate.Resources[naming.getMQEventLogicalId('other', queue)];
+      defaultIamRole = cfTemplate.Resources.IamRoleLambdaExecution;
+    });
+
+    it('should correctly compile EventSourceMapping resource properties with minimal configuration', () => {
+      expect(minimalEventSourceMappingResource.Properties).to.deep.equal({
+        EventSourceArn: brokerArn,
+        SourceAccessConfigurations: [
+          {
+            Type: 'BASIC_AUTH',
+            URI: basicAuthArn,
+          },
+        ],
+        Queues: [queue],
+        FunctionName: {
+          'Fn::GetAtt': [naming.getLambdaLogicalId('foo'), 'Arn'],
+        },
+      });
+    });
+
+    it('should update default IAM role with DescribeBroker statement', () => {
+      expect(defaultIamRole.Properties.Policies[0].PolicyDocument.Statement).to.deep.include({
+        Effect: 'Allow',
+        Action: ['mq:DescribeBroker'],
+        Resource: [brokerArn],
+      });
+    });
+
+    it('should update default IAM role with SecretsManager statement', () => {
+      expect(defaultIamRole.Properties.Policies[0].PolicyDocument.Statement).to.deep.include({
+        Effect: 'Allow',
+        Action: ['secretsmanager:GetSecretValue'],
+        Resource: [basicAuthArn],
+      });
+    });
+
+    it('should correctly compile EventSourceMapping resource DependsOn ', () => {
+      expect(minimalEventSourceMappingResource.DependsOn).to.equal('IamRoleLambdaExecution');
+      expect(allParamsEventSourceMappingResource.DependsOn).to.equal('IamRoleLambdaExecution');
+    });
+
+    it('should correctly compile EventSourceMapping resource with all parameters', () => {
+      expect(allParamsEventSourceMappingResource.Properties).to.deep.equal({
+        EventSourceArn: brokerArn,
+        BatchSize: batchSize,
+        Enabled: enabled,
+        SourceAccessConfigurations: [
+          {
+            Type: 'BASIC_AUTH',
+            URI: basicAuthArn,
+          },
+        ],
+        Queues: [queue],
+        FunctionName: {
+          'Fn::GetAtt': [naming.getLambdaLogicalId('other'), 'Arn'],
+        },
+      });
+    });
+  });
+
+  describe('configuring mq events', () => {
+    it('should update default IAM role with EC2 statement', async () => {
```

Comment:
Is there a reason why this test is nested in this describe instead of being performed against outputs from `before` in `when there are mq events defined` describe?

Reply:
Good catch! I will refactor that.